### DISCUSSION TITLE
How to get a notification when a video frame I sent gets lost?

### DISCUSSION THREAD

**Discussion by lostmsu** [sipsorcery-org/sipsorcery#1012] [Q&A]
I am streaming to browser clients using SIPSorcery without key frames for lower bandwidth and latency. From looking at the issues it sounds like I need to handle FIR and/or picture loss feedback messages to force key frames (or retransmit). But how do I do that? How do I subscribe to receive the feedback messages?

Also, I noticed `IVideoSource` interface. If I implement it, will SIPSorcery handle above for me? I could not find `ForceKeyFrame` calls that would seem to be doing that in the source code.

**Comment by imtadashi** [comment]
Hi @lostmsu, in this example [https://github.com/sipsorcery-org/sipsorcery/tree/master/examples/WebRTCExamples/WebRTCtoFfplay](https://github.com/sipsorcery-org/sipsorcery/tree/master/examples/WebRTCExamples/WebRTCtoFfplay) key frame will be used to stream in FFplay. Referring this might help you.
